pkg/i2gw/ingress2gateway.go

```diff
@@ -90,6 +133,29 @@ func readProviderResourcesFromFile(ctx context.Context, providerByName map[Provi
 	return nil
 }
 
+func readGenericResourcesFromFile(inputFile, namespace string) ([]client.Object, error) {
+	objects := make([]client.Object, 0)
+	stream, err := os.ReadFile(inputFile)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read file %v: %w", inputFile, err)
+	}
+
+	unstructuredObjects, err := ExtractObjectsFromReader(bytes.NewReader(stream), namespace)
+	if err != nil {
+		return nil, fmt.Errorf("failed to extract objects: %w", err)
+	}
+	for _, o := range unstructuredObjects {
+		if o.GetNamespace() != namespace {
+			continue
+		}
+		if _, ok := FilteredResources[o.GetObjectKind().GroupVersionKind().GroupKind()]; ok {
+			continue
```

Comment:
Trying to think if there is a different way to do it. I less like the fact that every provider needs to register its resources and that we depends on it.. I have not thought on anything smart yet. 
But.. we should have all the resources the provider cares about (which are the resources that we need to filter out from the file) in its `storage`.. Maybe we can leverage that?

Reply:
I guess we can add a `GetCRDs()` method to the `provider` interface. We have access to all the providers in `i2gw` package, so we can do the registration in this package. Instead of having each one registering his own CRDs

Reply:
I followed @levikobi's suggestion and added a `GetCRDs` method to the provider interface. Let me know if it looks better now. In that case, I'll add documentation about the new method for the providers to implement.